source/Octo.Tests/Commands/CommandDesignFixture.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Cli.Commands;
+using Octopus.Cli.Infrastructure;
+
+namespace Octo.Tests.Commands
+{
+    [TestFixture]
+    public class CommandDesignFixture
+    {
+        [Test]
+        public void ShouldAttachWithCorrectAttribute()
+        {
+            var commandClassNamesWithoutCorrectAttribute = Assembly.GetAssembly(typeof(ICommand)).GetTypes()
+                .Where(t => typeof(ICommand).IsAssignableFrom(t) &&
+                            t.GetCustomAttribute<CommandAttribute>() == null &&
+                            !t.IsAbstract &&
+                            !t.IsInterface)
+                .Select(t => t.Name)
+                .ToList()
+                .AsReadOnly();
+
+            commandClassNamesWithoutCorrectAttribute.Should().BeEmpty($"Each command which implements '{nameof(ICommand)}' interface must be attached with '{nameof(CommandAttribute)}'. " +
+                                                                      $"The following command classes: ({string.Join(", ", commandClassNamesWithoutCorrectAttribute)}) are not good enough for you?");
```

Comment:
I don't understand what "not good enough for you?" means

Reply:
I guess, this is just to remind people not to leave a `read-to-use` command that is not decorated with the expected `CommandAttribute`, or remove the command attribute without knowing and ship it to production. Maybe just `require your attention`?

Reply:
"The type X implements ICommand but is not decorated with a CommandAttribute, which is required for the program to detect it as an available command"

Reply:
Cool